### PR TITLE
feat: support config file for tensorboard [DET-3900]

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
Allow a yaml file to be passed as an argument into the tensorboard start command, providing the option of running tensorboard in different container types. Additionally adds the `config` arg to print the config of a specified tensorboard.

## Test Plan
Tested manually. Tensorboard uses a cpu-only container by default, so specify a config file that uses a different container type (e.g. a gpu container). Use `det tensorboard config tensorboard_id` to verify that the specified container is being used.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: cli/determined_cli/tensorboard.py [hunk 1]
```diff
@@ -37,8 +37,13 @@ def start_tensorboard(args: Namespace) -> None:
...
 
-    req_body = {"trial_ids": args.trial_ids, "experiment_ids": args.experiment_ids}
-    resp = api.post(args.master, "tensorboard", body=req_body)
+    config = parse_config(args.config_file, None, [], [])
+    req_body = {
+        "config": config,
+        "trial_ids": args.trial_ids,
+        "experiment_ids": args.experiment_ids,
+    }
+    resp = api.post(args.master, "tensorboard", body=req_body).json()
```
Comment (neilconway) [inline cli/determined_cli/tensorboard.py:46]: Seems like we're calling `json()` both here and below?

**Comment by eecsliu** [comment]
This is my bad, I incorrectly moved work between branches.

File: docs/how-to/tensorboard.txt [hunk 2]
```diff
@@ -51,6 +51,20 @@ TensorBoard for multiple experiments use
   metrics from persistent storage. It may take up to 5 minutes for TensorBoard
   to receive data and render visualizations.
 
+Specifying a Command Config
```
Comment (neilconway) [inline docs/how-to/tensorboard.txt:54]: "Customizing TensorBoards" or something similar?

File: docs/how-to/tensorboard.txt [hunk 3]
```diff
@@ -51,6 +51,20 @@ TensorBoard for multiple experiments use
...
+---------------------------
+
+Determined supports initializing TensorBoard with a config file. This
+feature can be useful for running TensorBoard with a specific container
+image or with a different resource allocation. Configuration settings
+can be found in the  :ref:`command-notebook-configuration`.
+
+To launch Tensorboard with a config file, use
+``det tensorboard start <experiment-id> --config-file=my_config.yaml``.
```
Comment (neilconway) [inline docs/how-to/tensorboard.txt:63]: Add an example of a configuration customization we expect to be somewhat common?

File: docs/how-to/tensorboard.txt [hunk 4]
```diff
@@ -51,6 +51,20 @@ TensorBoard for multiple experiments use
   metrics from persistent storage. It may take up to 5 minutes for TensorBoard
   to receive data and render visualizations.
 
+Specifying a Command Config
+---------------------------
+
+Determined supports initializing TensorBoard with a config file. This
+feature can be useful for running TensorBoard with a specific container
+image or with a different resource allocation. Configuration settings
+can be found in the  :ref:`command-notebook-configuration`.
```
Comment (neilconway) [inline docs/how-to/tensorboard.txt:60]: Can we update the command-notebook-configuration doc to specify how that configuration can be applied to TensorBoards as well?

Comment (eecsliu) [reply docs/how-to/tensorboard.txt]: I added a clarification in the command-notebook-configuration, but everything there should be compatible with tensorboard.

**Pull request merged**